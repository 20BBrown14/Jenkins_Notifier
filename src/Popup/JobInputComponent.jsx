import React from 'react';
import InputField from 'terra-form-input/lib/InputField';
import PropTypes from 'prop-types';

const propTypes = {
/** Error message */
  errorMessage: PropTypes.string.isRequired,
  /** Whether form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for validation */
  handleValidation: PropTypes.func.isRequired,
};

const defaultProps = {
  formInvalid: false,
};

const JobInputComponent = (props) => {
  const {
    errorMessage,
    formInvalid,
    handleValidation,
  } = props;

  return (
    <InputField
      inputId="jobNameInput"
      label="Input url of job to track"
      error={errorMessage}
      help="With or without 'api/xml' appended"
      isInvalid={formInvalid}
      onChange={handleValidation}
    />
  );
};

JobInputComponent.propTypes = propTypes;
JobInputComponent.defaultProps = defaultProps;

export default JobInputComponent;
