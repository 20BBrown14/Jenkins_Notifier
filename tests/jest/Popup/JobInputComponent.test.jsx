import React from 'react';
import JobInputComponent from '../../../src/Popup/JobInputComponent';

describe('JobInputComponent', () => {
  it('should render a default view', () => {
    const testComponent = (shallow(<JobInputComponent errorMessage="someMessage" handleValidation={() => {}} />));
    expect(testComponent).toMatchSnapshot();
  });
});
