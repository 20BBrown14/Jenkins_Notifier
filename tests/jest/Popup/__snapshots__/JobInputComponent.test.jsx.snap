// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobInputComponent should render a default view 1`] = `
<InputField
  error="someMessage"
  errorIcon={
    <IconError
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
  help={null}
  hideRequired={false}
  inputAttrs={Object {}}
  inputId="jobNameInput"
  isInline={false}
  isInvalid={false}
  isLabelHidden={false}
  label="Input URL of repo to track"
  labelAttrs={Object {}}
  onChange={[Function]}
  required={false}
  showOptional={false}
  style={
    Object {
      "margin": "25px",
    }
  }
  value=""
/>
`;
